src/pages/dashboard/dashboard.ts

```diff
@@ -0,0 +1,126 @@
+import { DateTime } from '../../shared/helpers/date-time';
+import { DeviceProvider } from '../../providers/device/device';
+import { Insomnia } from '@ionic-native/insomnia';
+import { ScreenOrientation } from '@ionic-native/screen-orientation';
+import { ExaminerRoleDescription } from '../../providers/app-config/constants/examiner-role.constants';
+
+interface DashboardPageState {
+  appVersion$: Observable<string>;
+  todaysDate$: Date | DateTime | string;
+}
+
+@IonicPage()
+@Component({
+  selector: 'page-dashboard',
+  templateUrl: 'dashboard.html',
+})
+export class DashboardPage extends PracticeableBasePageComponent {
+
+  static readonly welshLanguage: string = 'Cymraeg';
+
+  @ViewChild(Navbar) navBar: Navbar;
+
+  pageState: DashboardPageState;
+  licenseProvided: boolean;
+  healthDeclarationAccepted: boolean;
+  subscription: Subscription;
+  inputSubscriptions: Subscription[] = [];
+  isBookedInWelsh: boolean;
+  conductedLanguage: string;
+  merged$: Observable<boolean | string>;
+  employeeId: string;
+  name: string;
+  role: string;
+
+  constructor(
+    store$: Store<StoreModel>,
+    public navController: NavController,
+    public navParams: NavParams,
+    public platform: Platform,
+    public authenticationProvider: AuthenticationProvider,
+    private deviceAuthenticationProvider: DeviceAuthenticationProvider,
+    public alertController: AlertController,
+    public appConfigProvider: AppConfigProvider,
+    private dateTimeProvider: DateTimeProvider,
+    private deviceProvider: DeviceProvider,
+    public screenOrientation: ScreenOrientation,
+    public insomnia: Insomnia,
+  ) {
+    super(platform, navController, authenticationProvider, store$);
+    this.employeeId = this.authenticationProvider.getEmployeeId() || 'NOT_KNOWN';
+    this.name = this.authenticationProvider.getEmployeeName() || 'Unknown Name';
+    this.role = ExaminerRoleDescription[this.appConfigProvider.getAppConfig().role] || 'Unknown Role';
+  }
+
+  ionViewDidEnter(): void {
+    this.store$.dispatch(new DashboardViewDidEnter());
+    this.navBar.backButtonClick = (e: UIEvent) => {
+      this.clickBack();
+    };
+    if (super.isIos()) {
+      this.screenOrientation.unlock();
+      this.insomnia.allowSleepAgain();
+      this.deviceProvider.disableSingleAppMode();
+    }
+  }
+
+  clickBack(): void {
+    this.deviceAuthenticationProvider.triggerLockScreen()
+      .then(() => {
+        this.navController.pop();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  ngOnInit(): void {
+    this.pageState = {
+      appVersion$: this.store$.pipe(
+        select(getAppInfoState),
+        map(getVersionNumber),
+      ),
+      todaysDate$: this.dateTimeProvider.now().format('dddd Do MMMM YYYY'),
+    };
+  }
+
+  ionViewWillEnter(): boolean {
+    if (this.merged$) {
+      this.subscription = this.merged$.subscribe();
+    }
+
+    return true;
```

Comment:
Won't need this return and the return type of the function is `void`

Reply:
The dashboard extends from the base page, and the `ionViewWillEnter` function on that component returns a boolean